packages/orchestrator-ui-components/src/rtk/endpoints/subscriptionInUseByRelationsList.ts

```diff
@@ -0,0 +1,54 @@
+import { CacheTags, orchestratorApi } from '@/rtk';
+import { InUseByRelationDetail, InUseByRelationsDetailResult } from '@/types';
+
+export const subscriptionInUseByRelationQuery = `
+    query SubscriptionsInUseByRelationsDetails(
+        $subscriptionIds: String!
+    ) {
+        subscriptions(
+            first: 1000000
+            after: 0
+            filterBy: [{field: "subscriptionId", value: $subscriptionIds},{field: "status", value: "INITIAL|ACTIVE|MIGRATING|DISABLED|PROVISIONING"}]
+          ) {
+            page {
+              product {
+                name
+              }
+              description
+              subscriptionId
+              status
+            }
+          }
+    }
+`;
+
+export type InUseByRelationsDetailResponse = {
+    inUseByRelationDetails: InUseByRelationDetail[];
+};
+
+const subscriptionInUseByRelationsListApi = orchestratorApi.injectEndpoints({
+    endpoints: (builder) => ({
+        getInUseByRelationDetails: builder.query<
+            InUseByRelationsDetailResponse,
+            { subscriptionIds: string }
+        >({
+            query: (variables) => ({
+                document: subscriptionInUseByRelationQuery,
+                variables,
+            }),
+            transformResponse: (
+                response: InUseByRelationsDetailResult,
```

Comment:
I still think this defeats the purpose of graphql, adding types to each and every query result since the whole point is being flexible. Feel free to ignore my ranting about typing in front-ends ;)